modules/text/src/text_detector_swt.cpp

```diff
@@ -0,0 +1,812 @@
+    std::vector<SWTPoint> points;
+};
+
+struct Component {
+    SWTPoint BB_pointP;
+    SWTPoint BB_pointQ;
+    float cx;
+    float cy;
+    float median;
+    float mean;
+    int length, width;
+    std::vector<SWTPoint> points;
+};
+
+struct ComponentAttr {
+    float mean, variance, median;
+    int xmin, ymin;
+    int xmax, ymax;
+    float length, width;
+};
+
+struct ChannelAverage {
+    float Red, Green, Blue;
+};
+
+struct Direction {
+    float x, y;
+};
+
+struct ChainedComponent {
+    int chainIndexA;
+    int chainIndexB;
+    std::vector<int> componentIndices;
+    float chainDist;
+    Direction dir;
+    bool merged;
+};
+
+const Scalar BLUE (255, 0, 0);
+const Scalar GREEN(0, 255, 0);
+const Scalar RED  (0, 0, 255);
+void addEdge(std::vector<int> adj[], int u, int v);
+void DFSUtil(int v, bool visited[], std::vector<int> adj[], int label, std::vector<int> &component_id);
+int connected_components(std::vector<int> adj[], std::vector<int> &component_id, int num_vertices);
+void SWTFirstPass (Mat edgeImage, Mat gradientX, Mat gradientY, bool dark_on_light, Mat & SWTImage, std::vector<Ray> & rays);
+bool sortBySWT (const SWTPoint &lhs, const SWTPoint &rhs);
+void SWTSecondPass (Mat & SWTImage, std::vector<Ray> & rays);
+void normalizeAndScale (const Mat& SWTImage, Mat& output);
+std::vector<std::vector<SWTPoint>> getComponents (Mat& SWTImage);
+ComponentAttr getAttributes(vector<SWTPoint> component, Mat& SWTImage);
+void renderComponents (const Mat& SWTImage, std::vector<Component> components, Mat& output);
+std::vector<Component> filterComponents(Mat& SWTImage, std::vector<std::vector<SWTPoint>> components, bool skipChecks);
+void renderComponentBBs (std::vector<Component> components, Mat& output);
+vector<cv::Rect> findValidChains(Mat input_image, Mat SWTImage, std::vector<Component> components, Mat & output, std::vector<cv::Rect> & chainedTextRegions);
+vector<cv::Rect> getComponentBBs (std::vector<Component> components);
+bool chainSortDist (ChainedComponent Chainl, ChainedComponent Chainr);
+bool chainSortLength (ChainedComponent Chainl, ChainedComponent Chainr);
+
+
+// A utility function to add an edge in an
+// undirected graph.
+void addEdge(std::vector<int> adj[], int u, int v) {
+    adj[u].push_back(v);
+    adj[v].push_back(u);
+}
+
+void DFSUtil(int v, bool visited[], std::vector<int> adj[], int label, std::vector<int> &component_id)
+{
+    // Mark the current node as visited and label it as belonging to the current component
+    visited[v] = true;
+    component_id[v] = label;
+    // Recur for all the vertices
+    // adjacent to this vertex
+    for (unsigned int i = 0; i < adj[v].size(); i++) {
+        int neighbour = adj[v][i];
+        if (!visited[neighbour]) {
+            DFSUtil(neighbour, visited, adj, label, component_id);
+        }
+    }
+}
+
+int connected_components(std::vector<int> adj[], std::vector<int> &component_id, int num_vertices) {
+    bool *visited = new bool[num_vertices];
```

Comment:
Manual memory managements must be avoided - new/delete, malloc/free.

Use C++ `std::vector` or OpenCV's `cv::AutoBuffer` instead.